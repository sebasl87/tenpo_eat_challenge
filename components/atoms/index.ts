export { default as RestaurantsLayout } from './RestaurantsLayout';
