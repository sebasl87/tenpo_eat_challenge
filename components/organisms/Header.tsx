import React from 'react'
import { Image, Text, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.header}>
                <Image
                    source={require("../../assets/imgs/user.png")}
                    style={styles.userImg}
                />
                <Image
                    source={require("../../assets/icons/search.png")}
                    style={styles.searchImg}
                />
            </View>
            <View style={styles.header}>
                <View style={{ flexDirection: "column" }}>
                    <Image
                        source={require("../../assets/imgs/logo.png")}
                        style={styles.logoImg}
                    />
                    <Text>DELIVER APP</Text>
                </View>
                <View style={{ flexDirection: "row", marginBottom: -16 }}>
                    <Image
                        source={require("../../assets/imgs/hand.png")}
                        style={styles.handImg}
                    />
                    <View style={{ top: "100%", height: 6 }}>

                        <Image
                            source={require("../../assets/icons/ellipse.png")}
                            style={styles.ellipseImg}
                        />
                    </View>
                </View>
            </View>
        </>
    );
}

export default Header