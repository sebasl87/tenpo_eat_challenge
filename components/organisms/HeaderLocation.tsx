import React from 'react'
import { Image, Text, TextInput, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.headerLocation}>
                <View style={{ marginBottom: 32, display: "flex", flexDirection: "row" }}>
                    <Image
                        source={require("../../assets/icons/location.png")}
                        style={{ height: 24, width: 24, marginRight: 6, paddingTop: 30 }}
                    />
                    <Text style={styles.fontTitle}>Agregar dirección de la entrega</Text>
                </View>
            </View>
            <TextInput placeholder='Escribe tu dirección' style={styles.inputLocation} />
        </>
    );
}

export default Header