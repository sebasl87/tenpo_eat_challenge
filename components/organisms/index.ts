export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
