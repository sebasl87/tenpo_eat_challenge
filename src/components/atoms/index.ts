import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
