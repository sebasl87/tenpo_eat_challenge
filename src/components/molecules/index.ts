export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
