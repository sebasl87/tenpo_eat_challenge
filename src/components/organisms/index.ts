export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
