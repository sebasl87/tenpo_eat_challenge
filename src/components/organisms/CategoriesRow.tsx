import React from 'react'
import { View, ScrollView, Text } from 'react-native';
import styled from 'styled-components/native';
import { CategoriesLayout } from '../atoms'

export interface ICategories { }

const Title = styled.Text`
  color: #333333;
  font-size: 18px;
  margin-bottom: 6px;
  margin-top: 32px;
`;

const linksData = [
    {
        uri: require("../../../assets/imgs/hamburguer.png"),
        category: "HAMBURGUESAS"
    },
    {
        uri: require("../../../assets/imgs/italian.png"),
        category: "ITALIANA"
    },
    {
        uri: require("../../../assets/imgs/pizza.png"),
        category: "PIZZAS"
    },

]

const CategoriesRow: React.FC<ICategories> = () => (
    <>

        <Title style={{ fontFamily: "Gotham-Bold" }}>CATEGORIAS</Title>
        <ScrollView horizontal={true}>
            <View style={{ flexDirection: "row" }}>
                {linksData.map((item, i) => (
                    <View key={i} style={{ marginRight: 6 }}>

                        <CategoriesLayout image={item.uri}>{item.category}</CategoriesLayout>
                    </View>
                ))}
            </View >
        </ScrollView>

    </>
);


export default CategoriesRow;