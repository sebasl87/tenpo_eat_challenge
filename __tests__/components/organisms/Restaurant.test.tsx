import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { Restaurant } from "../../../src/components/organisms";

export const mock_resto = {
    id: 1,
    uri: "https://i.imgur.com/1PLcyBL.png",
    uri_large: "https://i.imgur.com/25zvKsX.png",
    discount: "50",
    restaurantName: "Mcdonalds",
    rate: "3.5",
    lapseTime: "10-50",
    bgColor: ""
}

describe("Restaurant component", () => {
    it("should render successfully with Restaurant_mock", () => {
        const restaurantSection = renderer
            .create(<Restaurant resto={mock_resto} />)
            .toJSON();

        expect(restaurantSection).toMatchSnapshot();
    });
});
