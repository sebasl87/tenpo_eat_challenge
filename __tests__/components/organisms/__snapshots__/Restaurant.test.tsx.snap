// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant component should render successfully with Restaurant_mock 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 12,
          }
        }
      >
        <RCTScrollView>
          <View>
            <View
              style={
                [
                  {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "minWidth": "100%",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="clickMe"
              >
                <View
                  bgColor=""
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "",
                        "borderRadius": 16,
                        "display": "flex",
                        "height": 148,
                        "justifyContent": "center",
                        "marginTop": 10,
                        "overflow": "hidden",
                        "width": 148,
                      },
                    ]
                  }
                >
                  <Image
                    source={
                      {
                        "uri": "https://i.imgur.com/25zvKsX.png",
                      }
                    }
                    style={
                      [
                        {
                          "height": 148,
                          "width": 148,
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "minWidth": "100%",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#333333",
                      "fontSize": 22,
                      "marginTop": 8,
                    },
                    {
                      "fontFamily": "Gotham-Bold",
                    },
                  ]
                }
              >
                Mcdonalds
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "minWidth": "100%",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#adadad",
                      "fontSize": 12,
                      "marginTop": 8,
                    },
                    {
                      "fontFamily": "Gotham-Book",
                    },
                  ]
                }
              >
                Pizzeria - Snacks
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 200,
                    "minWidth": "100%",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#adadad",
                      "fontSize": 20,
                      "marginTop": 8,
                    },
                    {
                      "fontFamily": "Gotham-Book",
                    },
                  ]
                }
              >
                Detalle de restaurante
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Title component should render successfully with title_mock 1`] = `
<Text
  style={
    [
      {
        "color": "#333333",
        "fontFamily": "Gotham-Bold",
        "fontSize": 18,
        "marginBottom": 6,
      },
    ]
  }
>
  title_mock
</Text>
`;
